### ISSUE TITLE
Remove `replace metadata_exchange` from go.mod

### ISSUE THREAD

**Issue by mandarjog** [istio/istio#20395]
go.mod contains 
```
replace github.com/istio.io/proxy/src/envoy/tcp/metadata_exchange/config v0.0.0 => ./pilot/pkg/metadata_exchange
```

This should be removed when envoy-wasm is rebased on Envoy 1.13 and makes its way to istio-proxy and istio.

**Comment by mandarjog** [comment]
@PiotrSikora we are awaiting this PR on 1.13 Envoy.

**Comment by mandarjog** [comment]
@kyessenov

**Comment by howardjohn** [comment]
This is blocking https://github.com/istio/istio/pull/20379

**Comment by gargnupur** [comment]
Closing as fixed in Master and Relase-1.5
